page.md

```diff
@@ -36,6 +43,68 @@ The `comments` attribute of the `<pg:AlternativeImage>` attribute should be used
 
-For such use cases, `<pg:AlternativeImage>` may be used as a child of `<pg:TextRegion>`, `<pg:TextLine>`, `<pg:Word>` or `<pg:Glyph>`.
+For such use cases, `<pg:AlternativeImage>` may be used as a child of
+`<pg:TextRegion>`, `<pg:TextLine>`, `<pg:Word>` or `<pg:Glyph>`.
+
+## Attaching text recognition results to elements
+
+A PAGE document can attach recognized text to different typographical units of
+a page at different levels, such as block (`<pg:TextRegion>`), line
+(`<pg:TextLine>`), word (`<pg:Word>`) or glyph (`<pg:Glyph>`).
+
+To attach recognized text to an element `E`, it must be encoded as
+`UTF-8` in a single `<pg:Unicode>` element `U` within a `<pg:TextEquiv>`
+element `T` of `E`.
+
+`T` must be the last element of `E`.
+
+Leading and trailing whitespace (`U+0020`, `U+000A`) in the content of a
+`<pg:Unicode>` is not significant and must be removed from the string by
+processors.
+
+To encode an actual space character at the start or end of the content
+`<pg:Unicode>`, use a non-breaking space `U+00A0`.
+
+## Text recognition confidence
+
+The confidence of the correctness of the text recognition results in a
+`<pg:TextEquiv>` can be expressed in an attribute `@conf` as a float value
+between `0` and `1`, where `0` means "certainly wrong" and `1` means "certainly
+correct".
+
+## Attaching multiple text recognition results to elements
+
+Alternative text recognition results can be expressed by using multiple
+`<pg:TextEquiv>` wherever a single `<pg:TextEquiv>` would be allowed. When
+using mulitple `<pg:TextEquiv>`, they each must have an attribute `@index` with
+an integer number unique per set of `<pg:TextEquiv>` that allows ranking them
+in order of preference. `@index` of the first (preferred) `<pg:TextEquiv>` must be
+the value `1`.
+
+## Consistency of text results on different levels
+
+Since text results can be defined on different levels and those levels can
+be nested, text results information is redundant. To avoid inconsistencies,
+the following assertions must be true:
+
+  * text of `<pg:Word>` must be equal to the text results of all `<pg:Glyph>`
+    contained within, concatenateb with empty string
+  * text of `<pg:TextLine>` must be equal to the text results of all
+    `<pg:Word>` contained  within, concatenated with a single space (`U+0020`).
+  * text of `<pg:TextRegion>` must be equal to the text results of all
+  * `<pg:TextLine>` contained within, concatenated with a newline (`U+000A`).
+
+**NOTE:** "Concatenation" means joining a list of strings with a separator, no
+separator is added to the start or end of the resulting string.
+
+If any of these assertions fails for a PAGE document, a processor must proceed
+with the text results at the lowest level provided.
+
+**Example:** `<pg:Word>` has text `Fool` but contains `<pg:Glyph>` whose
+text results, when concatenated, form the text `Foot`. The processor must proceed as if
+the `<pg:Word` had the text `Foot`.
```

Comment:
Different segmentations are an issue, c.f. https://github.com/OCR-D/spec/issues/72, but this proposal here is simpler.

First of all, it allows us to identify inconsistencies between the levels.

Secondly, if there are inconsistencies, it's always the lowest level that "wins".

IMHO, this should not happen at all and a processor should be expected to either produce consistent `TextEquiv` on all levels (processor changed words-> adapt lines, blocks) or delete those `TextEquiv` that cannot be kept consistent (processor changed words -> delete glyphs).

> Question: Why save a textequiv for each level if the textequivs are overwritten by the deeper level anyway?

Convenience I suppose. Since enforcing these consistency rules requires generating and comparing text anyway, we could probably prune the upper levels after processing and repopulate them before processing.

> What will happen if multiple textequivs exists? Must the indexes match?

Yes, that's a good point I'll add a note on that.

Reply:
> What will happen if multiple textequivs exists? Must the indexes match?

Updated, @VolkerHartmann please have a look at f94c9bb